koan/utils.py

```diff
@@ -339,49 +342,22 @@ def check_dist():
 
 def os_release():
     """
-    This code is borrowed from Cobbler and really shouldn't be repeated.
+    This code detects your os with the distro module and return the name and version. If it is not detected correctly it
+    returns "unkown" (str) and "0" (float).
+    @:returns tuple (str, float)
+        WHERE
+        str is the name
+        int is the version number
     """
-    if check_dist() == "redhat":
-        fh = open("/etc/redhat-release")
-        data = fh.read().lower()
-        if data.find("fedora") != -1:
-            make = "fedora"
-        elif data.find("centos") != -1:
-            make = "centos"
-        else:
-            make = "redhat"
-        release_index = data.find("release")
-        rest = data[release_index + 7:-1]
-        tokens = rest.split(" ")
-        for t in tokens:
-            try:
-                match = re.match('^\d+(?:\.\d+)?', t)
-                if match:
-                    return (make, float(match.group(0)))
-            except ValueError:
-                pass
-        raise KX("failed to detect local OS version from /etc/redhat-release")
-
-    elif check_dist() == "debian":
-        import lsb_release
-        release = lsb_release.get_distro_information()['RELEASE']
-        return ("debian", release)
-    elif check_dist() == "ubuntu":
-        version = subprocess.check_output(
-            ("lsb_release", "--release", "--short")).rstrip()
-        make = "ubuntu"
-        return (make, float(version))
-    elif check_dist() in ("suse", "opensuse"):
-        fd = open("/etc/SuSE-release")
-        for line in fd.read().split("\n"):
-            if line.find("VERSION") != -1:
-                version = line.replace("VERSION = ", "")
-            if line.find("PATCHLEVEL") != -1:
-                rest = line.replace("PATCHLEVEL = ", "")
-        make = "suse"
-        return (make, float(version))
+    distroname = distro.like()
+    version = distro.version()
+
+    allowed_distros = ["rhel", "centos", "fedora", "debian", "ubuntu", "suse"]
+
+    if distroname in allowed_distros:
```

Comment:
Same problem with the `like()` here. Debian doesn't have it. I think Ubuntu does… but you better check it.

Reply:
I just fired up an 18.04 container.  So for a recent Ubuntu it looks like this:

```
>>> distro.linux_distribution()
('Ubuntu', '18.04', 'Bionic Beaver')
>>> distro.like()
'debian'
>>> distro.name()
'Ubuntu'
>>> distro.id()
'ubuntu'
```